src/uu/wc/src/wc.rs

```diff
@@ -205,6 +232,47 @@ pub fn uu_app<'a>() -> App<'a> {
         )
 }
 
+fn inputs(matches: &ArgMatches) -> UResult<Vec<Input>> {
+    match matches.values_of_os(ARG_FILES) {
+        Some(os_values) => {
+            if matches.is_present(options::FILES0_FROM) {
+                return Err(WcError::FilesDisabled(
+                    "file operands cannot be combined with --files0-from".into(),
+                )
+                .into());
+            }
+
+            Ok(os_values
+                .map(|i| {
+                    if i == "-" {
+                        Input::Stdin(StdinKind::Explicit)
+                    } else {
+                        Input::Path(i.into())
+                    }
+                })
+                .collect())
+        }
+        None => match matches.value_of(options::FILES0_FROM) {
+            Some(files_0_from) => create_paths_from_files0(&files_0_from),
+            None => Ok(vec![Input::Stdin(StdinKind::Implicit)]),
+        },
+    }
+}
+
+fn create_paths_from_files0(files_0_from: &str) -> UResult<Vec<Input>> {
+    let mut paths = String::new();
+    if files_0_from == "-" {
+        io::stdin().lock().read_to_string(&mut paths)?;
+    } else {
+        File::open(files_0_from)?.read_to_string(&mut paths)?;
+    }
+
+    Ok(paths
+        .split_terminator('\0')
+        .map(|path| Input::Path(path.into()))
```

Comment:
One unhandled case is that one of the lines in the files could also be stdin. I tested it with this Python script:
```python
f = open("wc_test", "w")
f.write("Cargo.toml\0-")
```
And then
```
wc --files0-from wc_test
```

Reply:
Good catch!